site/components/TransactionsModal.js

```diff
@@ -0,0 +1,146 @@
+import useTranslation from 'next-translate/useTranslation'
+
+import { EtherscanLink } from './EtherscanLink'
+import JustifiedBetweenRow from './JustifiedBetweenRow'
+import Modal from './Modal'
+import SvgContainer from './svg/SvgContainer'
+import { useNumberFormat } from '../hooks/useNumberFormat'
+
+const TransactionModalRow = function ({ text, value, tipLink = '' }) {
+  return (
+    <JustifiedBetweenRow
+      keyComponent={
+        <p
+          className={`text-gray-350 text-sm ${
+            tipLink ? 'flex justify-bewtween cursor-pointer' : ''
+          }`}
+          onClick={() => tipLink && window.open(tipLink, '_blank')}
+        >
+          {text}
+          {tipLink && <SvgContainer className="ml-1" name="questionmark" />}
+        </p>
+      }
+      valueComponent={<p className="text-sm font-semibold">{value}</p>}
+    />
+  )
+}
+
+// eslint-disable-next-line complexity
+const TransactionsModal = function ({ transaction, modalIsOpen, closeModal }) {
+  const { t } = useTranslation('common')
+  const formatNumber = useNumberFormat()
+
+  const isConfirmed = transaction.transactionStatus === 'confirmed'
+  const isError =
+    transaction.transactionStatus === 'canceled' ||
+    transaction.transactionStatus === 'error'
+
+  return (
+    <Modal
+      className="relative flex flex-col w-full h-screen bg-white border-0 shadow-lg outline-none max-w-screen-sm focus:outline-none md:rounded-lg md:h-auto"
+      modalIsOpen={modalIsOpen}
+      onRequestClose={closeModal}
+    >
+      <div className="p-6">
+        {/* Header: operation name and closing "x" */}
+        <div className="border-b">
+          <button className="float-right" onClick={closeModal}>
+            <SvgContainer name="close" />
+          </button>
+          <p className="mb-2 font-bold text-left">
+            {t(`${transaction.operation}`)}
```

Comment:
Minor detail, but if operation is a string, the interpolation can be dropped

```suggestion
            {t(transaction.operation)}
```

Reply:
Changed in b73e76b.